File: temp/ComputerStore/library/src/Repositories/ClientRepository.cpp
#include "Repositories/ClientRepository.h"


void ClientRepository::add(Client_shared_ptr client) {
    clients.push_back(client);
}

void ClientRepository::remove(Client_shared_ptr client) {
    for (unsigned int i = 0; i < clients.size(); i++)
        if (clients[i] == client)
            clients.erase(clients.begin()+i);
}

std::vector<Client_shared_ptr> ClientRepository::getAll() {
    return clients;
}

std::string ClientRepository::toString() {
    std::string res;

    for (auto &&i : clients)
        res += i->toString() + "\n";

    return res;
}

File: src/main/java/model/repositories/ClientRepository.java
package model.repositories;

import controller.exceptions.RepositoryException;
import lombok.NoArgsConstructor;
import model.entities.Client;

import java.util.List;

@NoArgsConstructor
public class ClientRepository implements Repository<Client> {

    private List<Client> clients;

    public ClientRepository(List<Client> clients) {
        this.clients = clients;
    }

    @Override
    public void add(Client element) throws RepositoryException {
        for (Client client:clients) {
            if(client.equals(element))
                throw new RepositoryException("This client already exists");
        }
        this.clients.add(element);
    }

    @Override
    public void remove(Client element) throws RepositoryException {
        for (Client client:clients) {
            if(client.equals(element)) {
                this.clients.remove(element);
            }
        }
        throw new RepositoryException("This client doesn't exist");
    }

    @Override
    public List<Client> getAll() {
        return clients;
    }

    @Override
    public String toString() {
        return "ClientRepository{" +
                "clients=" + clients +
                '}';
    }
}
